### PR TITLE
Convert to/from `PlutusData` and `CborBytes`

### LINKED ISSUE #1001: Identical implementation for Plutus `serialiseData`
On babbage, we have `serialiseData` as an on-chain function, but it's common to use it off-chain as well (e.g. to serialise a message for a signature). It would be great to have this in CTL as well.

### PR DESCRIPTION
**Pull request by amirmrad**
Closes #1001 .

### Pre-review checklist

- [x] All code has been formatted using our config (`make format`)
- [x] Any new API features or modification of existing behavior is covered as defined in our [test plan](https://github.com/Plutonomicon/cardano-transaction-lib/blob/develop/doc/test-plan.md)
- [x] The changelog has been updated under the `## Unreleased` header, using the appropriate sub-headings (`### Added`, `### Removed`, `### Fixed`), and the links to the appropriate issues/PRs have been included

### CODE REVIEW

File: src/Serialization/PlutusData.purs [hunk 1]
```diff
@@ -68,6 +72,12 @@ packPlutusList :: Array T.PlutusData -> Maybe PlutusList
 packPlutusList = map (_packPlutusList containerHelper)
   <<< traverse convertPlutusData
 
+toCborBytes :: T.PlutusData -> Maybe CborBytes
+toCborBytes = map (wrap <<< wrap <<< _plutusDataToBytes) <<< convertPlutusData
+
+serialiseData :: forall a. ToData a => a -> Maybe CborBytes
```
Comment (ngua) [inline src/Serialization/PlutusData.purs:78]: Missing a kind signature here

File: src/Deserialization/PlutusData.purs [hunk 2]
```diff
@@ -68,6 +73,15 @@ convertPlutusInteger pd = T.Integer <$> do
 convertPlutusBytes :: PlutusData -> Maybe T.PlutusData
 convertPlutusBytes pd = T.Bytes <$> _PlutusData_bytes maybeFfiHelper pd
 
+fromCborBytes :: CborBytes -> Maybe T.PlutusData
+fromCborBytes = flip bind convertPlutusData
+  <<< _plutusDataFromBytes maybeFfiHelper
+  <<< unwrap
+  <<< unwrap
+
+deserialiseData :: forall a. FromData a => CborBytes -> Maybe a
```
Comment (ngua) [inline src/Deserialization/PlutusData.purs:82]: Missing kind signature

**Review by ngua — CHANGES_REQUESTED** [review]

**Review by klntsky — CHANGES_REQUESTED** [review]
Please use `Deserialization.FromBytes` machinery

**Review by klntsky — CHANGES_REQUESTED** [review]

File: src/Serialization/PlutusData.js [hunk 3]
```diff
@@ -33,3 +33,5 @@ exports._packMap = first => second => kvs => {
   }
   return res;
 };
+
+exports._plutusDataToBytes = data => data.to_bytes();
```
[the remark this answers is not in the document]
Comment (amirmrad) [reply src/Serialization/PlutusData.js:37]: will do

File: src/Serialization.purs [hunk 4]
```diff
@@ -891,3 +888,7 @@ hashScriptData cms rs ps = do
     [] -> _hashScriptDataNoDatums rs' cms'
     _ -> _hashScriptData rs' cms' =<< fromJustEff "failed to convert datums"
       (traverse convertPlutusData ps)
+
+serialiseData :: forall (a :: Type). ToData a => a -> Maybe CborBytes
```
Comment (klntsky) [inline src/Serialization.purs:892]: We need to re-export this function as part of `Contract` interface. I think `Contract.PlutusData` is a good place for it.

`deserializeData` should go there as well.

File: CHANGELOG.md [hunk 5]
```diff
@@ -94,6 +94,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - `PlutusV2.AlwaysSucceeds` example, for testing PlutusV2 scripts. ([#947](https://github.com/Plutonomicon/cardano-transaction-lib/pull/947))
 - `InlineDatum` example, for testing inline datum constraints. ([#931](https://github.com/Plutonomicon/cardano-transaction-lib/pull/931))
 - `DatumPresence` data type, which tags paying constraints that accept datum, to mark whether the datum should be inline or hashed in the transaction output. ([#931](https://github.com/Plutonomicon/cardano-transaction-lib/pull/931))
+- Added conversion functions between `PlutusData` and `CborBytes` ([#1001](https://github.com/Plutonomicon/cardano-transaction-lib/issues/1001))
```
Comment (klntsky) [inline CHANGELOG.md:97]: Please add names of the functions, with module names: `Contract.PlutusData.serializeData` and `Contract.PlutusData.deserializeData`

**Review by klntsky — CHANGES_REQUESTED** [review]

**Review by klntsky — APPROVED** [review]

**Pull request merged**